glue-codes/openfast-cpp/src/OpenFAST.H

```diff
@@ -22,53 +22,138 @@
 
 namespace fast {
 
-struct globTurbineDataType {
-  int TurbID;
-  std::string FASTInputFileName;
-  std::string FASTRestartFileName;
-  std::vector<double> TurbineBasePos;
-  std::vector<double> TurbineHubPos;
-  int numForcePtsBlade;
-  int numForcePtsTwr;
-  float nacelle_cd{0.0};
-  float nacelle_area{0.0};
-  float air_density{0.0};
+//! A data structure to hold all turbine related information
+struct turbineDataType {
+    //!Integer id for every turbine
+    int TurbID;
+    //! The FAST Input file name. Typically a .fst file.
+    std::string FASTInputFileName;
+    //! The restart/checkpoint file name. 
+    std::string FASTRestartFileName;
+    //! The position of the base of the turbine in global coordinates
+    std::vector<float> TurbineBasePos;
+    //! The approximate position of the hub of the turbine in global coordinates
+    std::vector<double> TurbineHubPos;
+    //! Number of blades 
+    int numBlades;
+    //! Number of velocity nodes (AeroDyn) per blade
+    int numVelPtsBlade;
+    //! Number of velocity nodes (AeroDyn) on the tower
+    int numVelPtsTwr;
+    //! Total number of velocity nodes (AeroDyn)
+    int numVelPts;
+    //! Desired number of actuator points on each blade
+    int numForcePtsBlade;
+    //! Desired number of actuator points on the tower
+    int numForcePtsTwr;
+    //! Total number of actuator points
+    int numForcePts;
+    //! Inflow Type - 1 (InflowWind) or 2 (Externally specified)
+    int inflowType;
+    //! Drag coefficient of nacelle
+    float nacelle_cd;
+    //! Frontal area of the nacelle
+    float nacelle_area;
+    //! Air density around this turbine
+    float air_density;
 };
 
+//! An id to indicate whether a particular actuator point is on the hub, node or tower    
 enum ActuatorNodeType {
   HUB = 0,
   BLADE = 1,
   TOWER = 2,
   ActuatorNodeType_END
 };
 
+/** An id to indicate the start type of a simulation.
+ * init - Start the simulation from scratch
+ * trueRestart - Restart from a checkpoint file. Code expects checkpoint files for all parts of the simulation including the controller.
+ * restartDriverInitFAST - Start all turbines from scratch and use the velocity data in 'velData.h5' file to run upto desired restart time, then continue the simulation like ''trueRestart'.
+ */
 enum simStartType {
```

Comment:
Might want to follow the convention of all caps for enum that you used for `ActuatorNodeType`

Reply:
Agreed. Implemented in 9f34f9eb119d38dc0c78bfe58fdd23d084d02430.